src/main/java/upb/dice/twitter/TweetSearchApplication.java

```diff
@@ -1,156 +1,77 @@
-            String line;
-            while ((line = br.readLine()) != null) {
-                String[] splitString = line.split(":");
-                String latitude = splitString[0];
-                String longitude = splitString[1];
-                String locMaxId = splitString[2];
-                pairStringMap.put(Arrays.asList(latitude, longitude), locMaxId);
-            }
-            String lon_str = String.valueOf(geoLocation.getLatitude());
-            String lat_str = String.valueOf(geoLocation.getLatitude());
-            if (pairStringMap.containsKey(Arrays.asList(lon_str, lat_str))) {
-                String id_string = pairStringMap.get(Arrays.asList(lon_str, lat_str));
-                return Long.parseLong(id_string);
-            }
-            return 1L;
-        }
-    }
-
-
-    private void queryBuild(String key) throws TwitterException, IOException {
-        long key_id_1 = keyParse(key);
-        Query query1 = new Query(key);
-        if (key_id_1 != 1) {
-            query1.sinceId(key_id_1);
-        }
-        getTweet(query1);
-    }
-
-    //Location based Tweet Search
-    private void queryBuild(double latitude, double longitude, double radius) throws TwitterException, IOException {
-        GeoQuery geoQuery = new GeoQuery(new GeoLocation(latitude, longitude));
-        Query query = new Query();
-        query.setGeoCode(geoQuery.getLocation(), radius, Query.KILOMETERS);
-        lat = latitude;
-        lon = longitude;
-
-//       query.until("2019-02-31").since("2019-02-01");   //Specific time period
-        long key_id_1 = locationParse(new GeoLocation(latitude, longitude));
-        if (key_id_1 != 1) {
-            query.sinceId(key_id_1);
-        }
-        getTweet(query);
-    }
-
-
-    //Extract the tweets based on the query built
-    private void getTweet(Query query) throws TwitterException, IOException {
-        QueryResult queryResult;
-        System.out.println("Tweets search Start");
-        FileWriter writer = new FileWriter("Tweets.txt", true);
-        int counter = 0;
-        do {
-            queryResult = twitter.search(query);
-            List<Status> queryTweets = queryResult.getTweets();
-            long maxId;
-            if (counter == 0) {
-                maxId = queryResult.getMaxId();
-                //Location based
-                if (query.getQuery() == null) {
-                    try (BufferedWriter out = new BufferedWriter(new FileWriter("Latitude_Longitude_MaxID.txt", true))) {
-                        out.append(String.valueOf(lat)).append(":").append(String.valueOf(lon)).append(":").append(String.valueOf(maxId));
-                        out.append('\n');
+        CommandLine finalCmd = cmd;
+        assert finalCmd != null;
+            timer.schedule(new TimerTask() {
+                @Override
+                public void run() {
+                    TweetExplorer searchTweets = new TweetExplorer();
+                    if (finalCmd.hasOption("h")) {
```

Comment:
I would suggest to handle the `"h"` option outside of the timer (i.e., before the timer is created).

Apart from that, `System.exit(0)` is something that should only be used in situations where it can not be avoided. Here, it looks like it is used just because it is easier than stopping all the running threads and exiting the normal way.